### DISCUSSION TITLE
(push-notifications/android) Open foreground push notification with context

### DISCUSSION THREAD

**Discussion by abennouna** [ionic-team/capacitor#6439] [Show and tell]
Currently in the Push Notifications plugin 4.1.2, the **foreground** notifications, i.e. the ones that are delivered while the app in active and in foreground, are shown in Android, but:

- do not disappear on tap
- do not trigger the `pushNotificationActionPerformed` listener
- the icon color cannot be customized

I succeeded in performing that 3 features that are needed for me anyway.

The result are here:

- to have tap makes foreground notif disappear --> https://github.com/abennouna/capacitor-plugins/commit/dabc08906549161740174b824b431eefaf84b276
- to customize icon color --> https://github.com/abennouna/capacitor-plugins/commit/c83206a11bcc7534794aadb353ff47b18f107045 (also see https://github.com/abennouna/capacitor-plugins/commit/a63d66db3211e7323a4c71541011d3e1071d0340 to have the same color for both background and foreground notifs)
- trigger the `pushNotificationActionPerformed` listener --> https://github.com/abennouna/capacitor-plugins/commit/d9ec714a5ba21079b6b7d56a504cac8bbd644ef5

I'm talking only about Android here. iOS works just fine as documented.

Do you think everyone would need the 3 features, or one of them? Should I open a PR for any?

**Comment by abennouna** [comment]
Update: https://github.com/abennouna/capacitor-plugins/commit/5f49e403522a42a7dfdd242b6d8c2bc0c3acc2fb

Two PRs already address the tap issue and `pushNotificationActionPerformed` issue:
- https://github.com/ionic-team/capacitor-plugins/pull/1423
- https://github.com/ionic-team/capacitor-plugins/pull/1478

There is also this older PR:
- https://github.com/ionic-team/capacitor-plugins/pull/1324

Well basically the solution has been around from some time.

**Comment by RaddishIoW** [comment]
I've published my own plugin after realising that the PRs are just being ignored.

It's a drop-in replacement for the official plugin. It's released as a beta at the mo, just because I've only quickly tested it with my project...
[@haylltd/capacitor-push-notifications](https://www.npmjs.com/package/@haylltd/capacitor-push-notifications)

I'd welcome any community PRs for extra features or bug fixes.

**Reply by facusatoshi** [reply]
Hi I will to test your version cause i need the features works on android.
Do you think that your pull requests will be approved?

Regards

**Reply by abennouna** [reply]
I settled on using patch-package instead of waiting for PRs to be merged, and instead of using another plugin.
